### PR TITLE
Batch of improvements

### PR DESCRIPTION
**Pull request by springuper**
main changes:

- rewrite js part of android lib and the js module, in order to use only one global variable and keep it more maintainable

i've make the js module compatible with Android lib.

TODO:

- [x] fix all FIXMEs
- [x] remove debugging code, specially, comments
- [x] support native->js async call, so that js handler could response to native after a while

### CODE REVIEW

File: dsbridge/DUIwebview.m [hunk 1]
```diff
@@ -107,7 +107,7 @@ - (void)webView:(UIWebView *)webView didCreateJavaScriptContext:(JSContext *)ctx
     ctx[@"_dsbridge"]=^(NSString * method,NSString * args){
         return [JSBUtil call:method :args JavascriptInterfaceObject:JavascriptInterfaceObject jscontext:ctx_];
     };
-    [ctx evaluateScript:INIT_SCRIPT];
+    // [ctx evaluateScript:INIT_SCRIPT];
```
Comment (springuper) [inline dsbridge/DUIwebview.m:110]: should we totally drop UIwebview since we only consider iOS 8+?

File: dsbridge/DWKwebview.m [hunk 2]
```diff
@@ -11,11 +11,13 @@
...
 @implementation DWKwebview
 {
-    void(^alertHandler)(void);
+    void (^alertHandler)(void);
     void (^confirmHandler)(BOOL);
     void (^promptHandler)(NSString *);
     int dialogType;
     UITextField *txtName;
+    int counter;
+    NSMutableDictionary *callbackDict;
```
Comment (springuper) [inline dsbridge/DWKwebview.m:20]: i don't know how to keep a callback dictionary, need some help.

`callbackDict` should have callbackId as its keys, and complete handlers as its values.

File: dsbridge/DUIwebview.m (hunk 1 above)
Comment (stevenojo) [reply dsbridge/DUIwebview.m:110]: Yeah, lets use WK web view

File: dsbridge/DWKwebview.m (hunk 2 above)
Comment (stevenojo) [reply dsbridge/DWKwebview.m:20]: That's the correct kind of object to store the dictionary of callbacks. 
Where would we need to set the values and retrieve them? 

The syntax to set would be [callbackDict setObject:callback forKey:calbackId];

Comment (springuper) [reply dsbridge/DWKwebview.m:20]: thanks! please help CR this new created commit: https://github.com/Tubitv/DSBridge-IOS/pull/1/commits/7adb440f06035a11b82006e87f1d645276868ca7.

**Review by stevenojo — APPROVED** [review]
LGTM. 
Great job on using the copy method on a block.

**Comment by springuper** [comment]
@stevenojo i refined some interfaces from json string to stricter `NSDictionary` to guarantee the return/callback value of ios handlers is always json object compatible. please check and tell me whether it's ok, you can first check the demo to feel it.

here it is: https://github.com/Tubitv/DSBridge-IOS/pull/1/commits/792663324302525bca765422497d9bc97bacb3b9.

**Pull request merged**